### DISCUSSION TITLE
Modeling a Curved Composite; Spatially Varying Orientation Field

### DISCUSSION THREAD

**Discussion by marc1russell** [idaholab/moose#27104] [Q&A Modules: Solid mechanics]
It seems possible to define orientations for entire blocks, but is there functionality to define an orientation field and enable orientation to vary element to element? Application could be modeling an orthotopic material with a complex geometry or a grain-microstructure.

**Comment by GiudGiud** [comment]
@dschwen @jiangwen84

**Comment by m1russell** [comment]
Perhaps a better question would be, what would be the best way to approach implementing such a functionality?

**Reply by GiudGiud** [reply]
An auxiliary variable that is of a vector type, for example a constant monomial_vec should do the trick 

there s work in the phase field module for orientations

**Comment by m1russell** [comment]
Thanks GiudGiud, that helped immensely.

It looks like the CP folks use PropertyReadFile within the ComputeElasticityTensorCP to read in Euler angles element by element from a txt file to rotate the element stiffness. Something similar should be possible for just a generic orthotopic material.

**Reply by GiudGiud** [reply]
ok that makes sense. The PropertyReadFile will let you specify 'nearest' (voronoi-) partitions, which can map to grains.
There's also an option for specifying data on a per-element basis but then you have to use the ids so your workflow for generating data must be well integrated with your meshing workflow

**Reply by m1russell** [reply]
That worked. I was able to use PropertyReadFile and store the rotation as a StatefulProperty to avoid reading at every time-step. Many thanks.

**Comment by m1russell** [comment]
> That worked. I was able to use PropertyReadFile and store the rotation as a StatefulProperty to avoid reading at every time-step. Many thanks.
